nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,799 @@
+
+    if (selector is not None) and do_screening:
+        X_train = selector.fit_transform(X[train], y[train])
+        X_test = selector.transform(X[test])
+
+    param_grid = _check_param_grid(estimator, X_train, y_train, param_grid)
+    best_score = None
+    for param in ParameterGrid(param_grid):
+        estimator = clone(estimator).set_params(**param)
+        estimator.fit(X_train, y_train)
+
+        if is_classif:
+            if hasattr(estimator, 'predict_proba'):
+                y_prob = estimator.predict_proba(X_test)
+                y_prob = y_prob[:, 1]
+                inverse_prob = 1 - y_prob
+            else:
+                decision = estimator.decision_function(X_test)
+                if decision.ndim == 2:
+                    y_prob = decision[:, 1]
+                    inverse_prob = np.abs(decision[:, 0])
+                else:
+                    y_prob = decision
+                    inverse_prob = -decision
+            score = scorer(estimator, X_test, y_test)
+            if np.all(estimator.coef_ == 0):
+                score = 0
+        else:  # regression
+            y_prob = estimator.predict(X_test)
+            score = scorer(estimator, X_test, y_test)
+
+        if (best_score is None) or (score >= best_score):
+            best_score = score
+            best_coef = estimator.coef_
+            best_intercept = estimator.intercept_
+            best_y = {}
+            best_y['y_prob'] = y_prob
+            best_y['y_true_indices'] = y_true_indices
+            best_param = param
+            if is_classif:
+                best_y['inverse'] = inverse_prob
+
+    if selector is not None:
+        best_coef = selector.inverse_transform(best_coef)
+
+    if LooseVersion(sklearn.__version__) <= LooseVersion('0.17'):
+        # scikit-learn > 0.17
+        if isinstance(estimator, SVR):
+            best_coef = -best_coef
+
+    return (class_index, best_coef, best_intercept, best_y, best_param,
+            best_score)
+
+
+class BaseDecoder(LinearModel, RegressorMixin, CacheMixin):
+    """A wrapper for popular classification/regression strategies for
+    neuroimgaging.
+
+    The `BaseDecoder` object supports classification and regression methods.
+    It implements a model selection scheme that averages the best models
+    within a cross validation loop. The resulting average model is the
+    one used as a classifier or a regressor. The `Decoder` object also
+    leverages the `NiftiMaskers` to provide a direct interface with the
+    nifti files on disk.
+
+    Parameters
+    -----------
+    estimator : str, optional
+        The estimator to choose among: 'svc', 'svc_l1', 'logistic',
+        'logistic_l1', 'ridge_classifier', 'ridge_regression',
+        and 'svr'. Defaults to 'svc'.
+
+    mask: filename, NiImage, NiftiMasker, or MultiNiftiMasker, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
```

Comment:
"then its mask and parameters will be used." I think it will be more clear.
Correct me if I am wrong here.